app_tests/integration_tests/llm/shortfin/tracing_test.py

```diff
@@ -0,0 +1,153 @@
+"""
+Integration test to ensure tracing functionality works correctly with the Shortfin LLM server.
+This tests that the server can generate a valid tracy capture file without crashing.
+"""
+
+import logging
+import os
+import pytest
+import requests
+import time
+import uuid
+import subprocess
+from pathlib import Path
+from typing import Dict, Any, Tuple
+
+from ..model_management import AccuracyValidationException
+from ..server_management import ServerInstance, ServerConfig
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.parametrize(
+    "model_artifacts,server",
+    [
+        ["tinystories_llama2_25m", {"prefix_sharing": "none"}],
+    ],
+    indirect=True,
+)
+
+# Same golden values as in tinystories_llama2_25m_test.py
+GOLDEN_PROMPT = "Once upon a time"
+GOLDEN_RESPONSE = ", there was a little girl named Lily."  # this assumes purely deterministic greedy search
+
+
+class TestTracingFunctionality:
+    """Test suite for tracing functionality with the LLM server."""
+
+    def test_tracing_generation(self, server: Tuple[Any, int], tmp_path: Path) -> None:
+        """Tests that tracing works while running text generation.
+
+        Args:
+            server: Tuple of (process, port) from server fixture
+            tmp_path: Temporary directory for storing the trace file
+        """
+        process, port = server
+        assert process.poll() is None, "Server process terminated unexpectedly"
+
+        # Determine output directory for artifacts (use environment variable if set, otherwise tmp_path)
+        output_dir = os.environ.get("TEST_OUTPUT_DIR")
+        if output_dir:
+            output_dir = Path(output_dir)
+            output_dir.mkdir(parents=True, exist_ok=True)
+        else:
+            output_dir = tmp_path
```

Comment:
Is this a common pattern in shark-ai? Python and pytest both provide ways to get temp paths that could be used instead